src/main/kotlin/pl/allegro/mobile/logic/operators/string/CompareToDateOperation.kt

```diff
@@ -0,0 +1,33 @@
+package pl.allegro.mobile.logic.operators.string
+
+import pl.allegro.mobile.logic.ClientLogicElement
+import pl.allegro.mobile.logic.ClientLogicMarker
+import pl.allegro.mobile.logic.ClientLogicOperator
+import pl.allegro.mobile.logic.StringElement
+
+
+internal interface  CompareToDateOperation {
+    /**
+     * Compares a string to given string represents the date in proper format.
+     * This operation compares the date
+     * @receiver Character sequence or client side operation that returns string
+     * @param date date to compare to date represents in current string
+     * @return integer interpretation of state. State is between -1 and 1.
+     * -1 - passing date is BEFORE the compared date
+     * 0 - passing date is EQUALS the compared date
+     * 1 - passing date is AFTER the compared date
```

Comment:
As I mentioned in the other PR: I'm wondering if it won't be better to switch `1` and `-1` values. E.g. kotlin's `compareTo(other: ...)` operator returns `1` when `this` is "bigger" and `-1` in the opposite, so maybe it is better to stick to this convention ([docs](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin/-int/compare-to.html)).